opentelekomcloud/services/apigw/data_source_opentelekomcloud_apigw_api_history.go

```diff
@@ -0,0 +1,120 @@
+package apigw
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+	apis "github.com/opentelekomcloud/gophertelekomcloud/openstack/apigw/v2/api"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/cfg"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/fmterr"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/helper/hashcode"
+)
+
+func DataSourceApigwApiHistory() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceEnvironmentsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
```

Comment:
`region` is not provided in `ListHistoryOpts` therefore should be moved to `computed` instead of `optional`